components/datadog/agentparams/filepermissions/permissions_nix.go

```diff
@@ -0,0 +1,82 @@
+package filepermissions
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+	"github.com/DataDog/test-infra-definitions/common"
+)
+
+type UnixPermissionsOption = func(*UnixPermissions) error
+
+// UnixPermissions represents the owner, group, and permissions of a file.
+// Permissions are represented by a string (should be an octal). There is no check on the octal format.
+type UnixPermissions struct {
+	Owner       string
+	Group       string
+	Permissions string
+}
+
+var _ FilePermissions = (*UnixPermissions)(nil)
+
+// NewUnixPermissions creates a new UnixPermissions object and applies the given options.
+func NewUnixPermissions(options ...UnixPermissionsOption) optional.Option[FilePermissions] {
+	p, err := common.ApplyOption(&UnixPermissions{}, options)
+
+	if err != nil {
+		return optional.NewNoneOption[FilePermissions]()
+	}
+	return optional.NewOption[FilePermissions](p)
+}
+
+// SetupPermissionsCommand returns a command that sets the owner, group, and permissions of a file.
+func (p *UnixPermissions) SetupPermissionsCommand(path string) string {
+	var commands []string
+
+	if p.Owner != "" {
+		commands = append(commands, fmt.Sprintf("sudo chown %s %s", p.Owner, path))
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I hope no one will have that, but if `path` contains spaces then the commands won't work properly 😄

Reply:
You're right, I quoted everything here 880744def3463c076ffbc447f5c1bd305a016a1e